clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4684,6 +4684,14 @@ void SYCLIntegrationHeader::emit(raw_ostream &O) {
 
   for (const KernelDesc &K : KernelDescs) {
     const size_t N = K.Params.size();
+    PresumedLoc PLoc = S.Context.getSourceManager().getPresumedLoc(
+        S.Context.getSourceManager()
+            .getExpansionRange(K.KernelLocation)
+            .getEnd());
+    std::string FileName = PLoc.getFilename();
```

Comment:
Since these are only used for the emit, I suggest not creating these temporary variables and just calling functions when emitting.

Reply:
PLoc is probably valid, and line/column number are innocuous and likely to be opted out, but you're absolutely right about the strings, particularly the KernelName, where 'getName' is emittable directly without calling .str.